echo_server.py

```diff
@@ -21,10 +24,49 @@ def echo_server():
                 if len(response) < 32:
                     done = True
                 text_received = "{}{}".format(text_received, response)
-            conn.sendall(text_received)
+            if text_received:
+                response_user = parase_request(text_received)
+            conn.sendall(response_user)
     except KeyboardInterrupt:
        server_socket.close()
 
+def response_ok(uri):
+    first_line = 'HTTP/1.1 200 OK'
+    timestamp = 'Date: ' + email.utils.formatdate(usegmt=True)
+    header = 'Content-Type: text/plain'
+    body = uri
+    size = str(sys.getsizeof(body))
+    end = '\r\n'
+    response  =  [first_line, timestamp, header, size, body, end]
```

Comment:
In this particular setup, your browser will not notice that the headers for this response have ended until that last `\r\n` at the end.  A proper HTTP response **must* have the first line, one or more headers, *a blank line* and then optionally a body to display in the browser window.

```
HTTP/1.1 200 OK
Content-Type: text/plain

/this/was/the/request/uri
```

Notice that the specifications for this stage of the server build did not indicate that `response_ok` would take any arguments.  That's because it was intended for you to practice building a completely static response piece by piece so you could concentrate on what that looks like.